libraries/AP_Scripting/AP_Scripting.h

```diff
@@ -56,6 +60,27 @@ class AP_Scripting
     };
     uint16_t get_disabled_dir() { return uint16_t(_dir_disable.get());}
 
+    struct mavlink_msg {
+        mavlink_message_t msg;
+        mavlink_channel_t chan;
+    };
+
+    struct mavlink_output {
+        char data[256]; // maximum payload size
+        int32_t msgid;
+        mavlink_channel_t chan;
+    };
+
+    static const int mavlink_input_queue_size = 5;
+    static const int mavlink_output_queue_size = 3;
+
+    struct mavlink {
+        ObjectBuffer<struct mavlink_msg> * input;
+        ObjectBuffer<struct mavlink_output> * output;
```

Comment:
much better to send direct from lua thread, with semaphore

Reply:
Done